cogs/TextFilter.py

```diff
@@ -0,0 +1,701 @@
+    :return: True if admin or test, False otherwise
+    """
+    try:
+        result = KoalaBot.check_guild_has_ext(ctx, "TextFilter")
+    except PermissionError:
+        result = False
+
+    return result or (str(ctx.author) == KoalaBot.TEST_USER and KoalaBot.is_dpytest)
+
+
+class TextFilter(commands.Cog, name="TextFilter"):
+    """
+    A discord.py cog with commands pertaining to the a Text Filter for admins to monitor their server
+    """
+
+    def __init__(self, bot, database_manager=None):
+        if not database_manager:
+            database_manager = KoalaBot.database_manager
+        self.bot = bot
+        database_manager.create_base_tables()
+        database_manager.insert_extension("TextFilter", 0, True, True)
+        self.tf_database_manager = TextFilterDBManager(database_manager, bot)
+        self.tf_database_manager.create_tables()
+
+    @commands.command(name="filter", aliases=["filter_word"])
+    @commands.check(KoalaBot.is_admin)
+    @commands.check(text_filter_is_enabled)
+    async def filter_new_word(self, ctx, word, filter_type="banned", too_many_arguments=None):
+        """
+        Adds a new word to the filtered text list
+        :param ctx: The discord context
+        :param word: The first argument and word to be filtered
+        :param filter_type: The filter type (banned or risky)
+        :param too_many_arguments: Used to check if too many arguments have been given
+        :return:
+        """
+        error = """Something has gone wrong, your word may already be filtered or you have entered the 
+                command incorrectly. Try again with: `k!filter [filtered_text] [[risky] or [banned]]`"""
+        if too_many_arguments is None and type_exists(filter_type):
+            await filter_text(self, ctx, word, filter_type, False)
+            await ctx.channel.send("*" + word + "* has been filtered as **"+filter_type+"**.")
+            return
+        raise Exception(error)
+
+    @commands.command(name="filterRegex", aliases=["filter_regex"])
+    @commands.check(KoalaBot.is_admin)
+    @commands.check(text_filter_is_enabled)
+    async def filter_new_regex(self, ctx, regex, filter_type="banned", too_many_arguments=None):
+        """
+        Adds a new regex to the filtered text list
+        :param ctx: The discord context
+        :param regex: The first argument and regex to be filtered
+        :param filter_type: The filter type (banned or risky)
+        :param too_many_arguments: Used to check if too many arguments have been given
+        :return:
+        """
+        error = """Something has gone wrong, this regex may already be filtered or you have entered the " 
+                command incorrectly. Try again with: `k!filterRegex [filtered_regex] [[risky] or [banned]]`. 
+                One example for a regex could be to block emails with: 
+                [a-z0-9]+[\._]?[a-z0-9]+[@]+[herts]+[.ac.uk] where EMAIL is the university type (e.g herts)"""
```

Comment:
Not all unis use this type of email structure, even though this may not be accurate, try changing it to:
```suggestion
                [a-zA-Z0-9\._]+@herts\.ac\.uk where EMAIL is the university type (e.g herts)"""
```
This should blanket most emails